tasks/utils/oauth.ts

```diff
@@ -0,0 +1,124 @@
+import { getConfig } from './config'
+import type { AuthorizationServer, Client, OpenIDTokenEndpointResponse } from 'oauth4webapi'
+import * as oauth from 'oauth4webapi'
+import {
+  authorizationCodeGrantRequest,
+  discoveryRequest,
+  generateRandomState,
+  isOAuth2Error,
+  processAuthorizationCodeOpenIDResponse,
+  processDiscoveryResponse,
+  validateAuthResponse
+} from 'oauth4webapi'
+
+const config = getConfig()
+const codeChallengeMethod = 'S256'
+
+const LOCALSTORAGE_KEY_STATE = 'oauth_state'
+const LOCALSTORAGE_KEY_CODE_VERIFIER = 'oauth_code_verifier'
```

Comment:
we should consider to use cookies instead (this depends on the importance of jwt here)